qubes/vm/qubesvm.py

```diff
@@ -216,6 +216,59 @@ def _setter_kbd_layout(self, prop, value):
     return value
 
 
+def _default_active_bootmode(self):
+    """
+    Return the template's default bootmode for AppVMs if possible, otherwise
+    return a blank string. May reset the template's AppVM default bootmode
+    property if the boot mode it specifies does not exist.
+    """
+    subject = self
+    while hasattr(subject, "template"):
+        if hasattr(subject.template, "appvm_default_bootmode"):
+            bootmode_value = subject.template.appvm_default_bootmode
+            kernelopts = subject.features.check_with_template(
+                f"boot-mode.{bootmode_value}.kernelopts", None
+            )
+            if kernelopts is None:
+                subject.template.appvm_default_bootmode = qubes.property.DEFAULT
+            else:
+                return bootmode_value
+        subject = subject.template
+    return ""
+
+
+def _setter_bootmode_kernelopts(self, prop, value):
+    """
+    Raises an exception, as bootmode_kernelopts should not be set.
+    """
+    raise qubes.exc.QubesPropertyValueError(
+        self, prop, value, "Cannot set bootmode_kernelopts directly"
+    )
+
+
+def _default_bootmode_kernelopts(self):
+    """
+    Return the kernel options specified by the currently active boot mode,
+    with a leading space. Reset the active_bootmode property if the specified
+    boot mode does not exist.
+    """
+    if self.active_bootmode == "":
```

Comment:
And same here about default boot mode name.

Reply:
Changed, the "no params" default boot mode is now `default`.